src/servers/main.cc

```diff
@@ -1154,6 +1154,10 @@ main(int argc, char** argv)
   auto shm_manager =
       std::make_shared<nvidia::inferenceserver::SharedMemoryManager>();
 
+  // Trap SIGINT and SIGTERM to allow server to exit gracefully
+  signal(SIGINT, SignalHandler);
```

Comment:
The problem with moving this here is that if you use Ctrl-C while loading is happening, then loading will continue until all the models are loaded... and then the endpoints will be started... and then triton will recognize the Ctrl-C and unload the models and exit. That means there can be a long delay between using Ctrl-C and triton actually exiting. Have you found a specific problem with the way things are currently handled.

Reply:
The problem is that if Ctrl-C is pressed before all the models are loaded, Triton will not trigger `ModelInstanceFinalize` and `ModelFinalize`. In the case of Python backend, this leads to not terminating GRPC servers for the models that have already been loaded.

Reply:
So what happens when triton exits non-gracefully?  Non-graceful exit needs to be handled because the sig handler can't catch all exits.